src/file/preader.cr

```diff
@@ -4,7 +4,11 @@ class File::PReader < IO
 
   getter? closed = false
 
-  def initialize(@file : File, @offset : Int32, @bytesize : Int32)
+  def self.new(file : File, offset : Int32, bytesize : Int32)
+    new file, offset.to_i64, bytesize.to_i64
+  end
+
+  def initialize(@file : File, @offset : Int64, @bytesize : Int64)
```

Comment:
Another way to do this, which is done everywhere else, is to restrict the argument type to `Int`, then call `to_i64` on that and store that as an instance variable. Then you can pass any int type.

Reply:
Wouldn't that make it seem like you could pass `UInt64` but then fail at runtime?

Reply:
Yes, but who uses `UInt64` anyways? And it will still fail at runtime so it's fine.

Reply:
And it will only fail with big values of `UInt64`. I don't see this as an issue, honestly.